src/ndarray/ndarray.cc

```diff
@@ -1214,6 +1214,41 @@ void NDArray::SyncCopyToCPU(void *data, size_t size) const {
   }
 }
 
+void NDArray::CheckFormat(const bool full_check) const {
+  int32_t err = kNormalErr;
+  TBlob err_cpu(&err, mshadow::Shape1(1), cpu::kDevMask, 0);
+  if (this->ctx().dev_mask() == cpu::kDevMask) {
+    Engine::Get()->PushSync([&](RunContext rctx) {
+        common::CheckFormatWrapper<cpu>(rctx, *this, err_cpu, full_check);
+      }, this->ctx(), {this->var()}, {},
+      FnProperty::kNormal, 0, PROFILER_MESSAGE("CheckFormat"));
+    this->WaitToWrite();
+  } else {
+#if MXNET_USE_CUDA
+    Engine::Get()->PushSync([&](RunContext rctx) {
+        common::CheckFormatWrapper<gpu>(rctx, *this, err_cpu, full_check);
+        rctx.get_stream<gpu>()->Wait();
+      }, this->ctx(), {this->var()}, {},
+      FnProperty::kNormal, 0, PROFILER_MESSAGE("CheckFormat"));
+    this->WaitToWrite();
+#else
+    LOG(FATAL) << "GPU is not enabled";
+#endif
+  }
+  CHECK_NE(err, kCSRShapeErr) << "Shape mismatch of this CSRNDArray";
+  CHECK_NE(err, kCSRIndPtrErr)
+           << "IndPtr of CSRNDArray should be non-negative, in non-decreasing order, "
+           << "start with 0, and end with value equal with size of indices.";
+  CHECK_NE(err, kCSRIdxErr)
+           << "Indices of CSRNDArray should be non-negative, in ascending order per row "
+           << " and less than the number of columns.";
+  CHECK_NE(err, kRSPShapeErr) << "Shape mismatch of this RSPNDArray";
+  CHECK_NE(err, kRSPIdxErr)
+          << "Indices of RSPNDArray should be non-negative, "
```

Comment:
We should avoid the usage of `RSPNDArray` in the log since it's displayed to the users. We didn't mention the term `RSPNDArray` in any other user facing documentation. This will also be used for all language bindings. Let's replace it with `row_sparse NDArray`/`csr NDArray` to avoid any python specific class names.